playground/vue-legacy/Main.vue

```diff
@@ -0,0 +1,32 @@
+<script>
+import { defineComponent } from 'vue'
+import css from './inline.css?inline'
+
+export default defineComponent({
+  components: {
+    module: defineAsyncComponent(() => import('./module.vue'))
+  },
+  setup() {
+    return {
+      css
+    }
+  }
+})
+</script>
+<template>
+  <div class="container">
+    <p>hello</p>
+    <module />
+    <code>
+      {{ css }}
+    </code>
+  </div>
+</template>
+<style scoped>
+.container {
+  height: 100vh;
+  background: url('@/assets/asset.png') no-repeat;
```

Comment:
When running in legacy (by doing the same thing in `__test__` function in `vite.config.ts`), it shows a blank page.

> Uncaught ReferenceError: module is not defined
 
The error above was happening.
It seems we are injecting `var __vite_style__ = document.createElement('style');` outside the SystemJS module. So we don't have access to `module.meta.url`.
```js
;

(function () {
  function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }

  function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }

  function _iterableToArrayLimit(arr, i) { var _i = arr == null ? null : typeof Symbol !== "undefined" && arr[Symbol.iterator] || arr["@@iterator"]; if (_i == null) return; var _arr = []; var _n = true; var _d = false; var _s, _e; try { for (_i = _i.call(arr); !(_n = (_s = _i.next()).done); _n = true) { _arr.push(_s.value); if (i && _arr.length === i) break; } } catch (err) { _d = true; _e = err; } finally { try { if (!_n && _i["return"] != null) _i["return"](); } finally { if (_d) throw _e; } } return _arr; }

  function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }

  function _createForOfIteratorHelper(o, allowArrayLike) { var it = typeof Symbol !== "undefined" && o[Symbol.iterator] || o["@@iterator"]; if (!it) { if (Array.isArray(o) || (it = _unsupportedIterableToArray(o)) || allowArrayLike && o && typeof o.length === "number") { if (it) o = it; var i = 0; var F = function F() {}; return { s: F, n: function n() { if (i >= o.length) return { done: true }; return { done: false, value: o[i++] }; }, e: function e(_e2) { throw _e2; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var normalCompletion = true, didErr = false, err; return { s: function s() { it = it.call(o); }, n: function n() { var step = it.next(); normalCompletion = step.done; return step; }, e: function e(_e3) { didErr = true; err = _e3; }, f: function f() { try { if (!normalCompletion && it.return != null) it.return(); } finally { if (didErr) throw err; } } }; }

  function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }

  function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) arr2[i] = arr[i]; return arr2; }

  var __vite_style__ = document.createElement('style');

  __vite_style__.innerHTML = "\n.container[data-v-d245be1d] {\n  height: 100vh;\n  background: url('" + new URL('asset.b9f46fb3.png', module.meta.url).href + "') no-repeat;\n  background-size: contain;\n  background-position: center;\n}\n";
  document.head.appendChild(__vite_style__);
  System.register([], function (exports, module) {
    'use strict';

    return {
```

Reply:
I thought that `import.meta.url` wouldn't be used after your PR https://github.com/vitejs/vite/pull/9381 🤔

Reply:
It's using `module.meta.url` which is equivalent to `import.meta.url` and provided by SystemJS, not `import.meta.url` itself.

The error is happening because we are using `module` variable outside `System.register` callback.

Reply:
Oh, sorry, I misread your prev comment. We could inject this snippet after `'use strict'` instead of at the top for System.js maybe

Reply:
can I use `location.href` here. Like the prev modify,

Reply:
After https://github.com/vitejs/vite/pull/9381, your new extracted function should use the right scheme depending on the chunk type. Is this not working if you inject it inside the System.js wrapper?